clusterloader2/pkg/provider/provider.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"fmt"
+	"strings"
+)
+
+type InitOptions struct {
+	KubemarkRootKubeConfigPath string
+	ProviderName               string
+	ProviderConfigs            []string
+}
+
+type Provider interface {
+	Name() string
+	CanEnablePrometheusServer() error
```

Comment:
Let's establish some naming convention here and name all the functions in the same way.
I propose to use: 
* `SupportsFeatureA() bool` for all feature gates. 
* `GetValueA()` for all provider specific values